plugins/flashrom/fu-flashrom-lspcon-i2c-spi-device.c

```diff
@@ -175,56 +210,80 @@ probe_active_flash_partition (FuFlashromLspconI2cSpiDevice *self,
-	flashrom_layout_set (flashctx, layout);
 
-	/* get the active partition */
+	/* determine active partition for flashing later */
 	if (!probe_active_flash_partition (self, &self->active_partition, error))
 		return FALSE;
 	g_debug ("device reports running from partition %d", self->active_partition);
 	if (self->active_partition < 1 || self->active_partition > 3) {
-		g_set_error (error, FWUPD_ERROR, FWUPD_ERROR_BROKEN_SYSTEM,
-			     "Unexpected active flash partition: %d",
+		g_set_error (error, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED,
+			     "unexpected active flash partition: %d",
 			     self->active_partition);
 		return FALSE;
 	}
 
-	/* read version bytes for the active partition from device flash */
-	layout->entries[self->active_partition + 2].included = TRUE;
-
-	/* read the current flash contents */
-	fu_device_set_status (device, FWUPD_STATUS_DEVICE_READ);
-	flash_size = fu_flashrom_device_get_flash_size (flashrom_device);
-	contents = g_malloc0 (flash_size);
-	if (flashrom_image_read (flashctx, contents, flash_size)) {
-		g_set_error_literal (error, FWUPD_ERROR, FWUPD_ERROR_READ,
-				     "failed to read flash contents");
+	/* find the drm_dp_aux_dev specified by quirks that is connected to the
+	 * LSPCON, in order to read DPCD from it */
+	if (self->aux_device_name == NULL) {
+		g_set_error_literal (error, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED,
+				     "no DP aux device specified, unable to query LSPCON");
+		return FALSE;
+	}
+	g_udev_enumerator_add_match_subsystem (enumerator, "drm_dp_aux_dev");
+	g_udev_enumerator_add_match_sysfs_attr (enumerator, "name", self->aux_device_name);
+	aux_devices = g_udev_enumerator_execute (enumerator);
+	if (aux_devices == NULL) {
+		g_set_error (error, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED,
+			     "failed to locate a DP aux device named \"%s\"",
+			     self->aux_device_name);
 		return FALSE;
 	}
+	if (g_list_length (aux_devices) > 1) {
+		g_list_free_full (aux_devices, g_object_unref);
+		g_set_error (error, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED,
+			     "found multiple DP aux devices with name \"%s\"",
+			     self->aux_device_name);
+		return FALSE;
+	}
+	aux_device = fu_udev_device_new (g_steal_pointer (&aux_devices->data));
+	g_list_free (aux_devices);
+	g_debug ("using aux dev %s", fu_udev_device_get_sysfs_path (aux_device));
+	/* open() requires the device have IDs set */
+	if (!fu_udev_device_set_physical_id (aux_device, "drm_dp_aux_dev", error))
+		return FALSE;
 
-	/* extract the active partition's version number */
-	addr = layout->entries[self->active_partition + 2].start;
-	g_return_val_if_fail (addr < (flash_size - 2), FALSE);
-	version = g_strdup_printf ("%d.%d", contents[addr], contents[addr + 2]);
+	/* DPCD address 0x50A, 0x50B: branch device firmware
+	 * major and minor revision */
+	if ((aux_device_locker = fu_device_locker_new (aux_device, error)) == NULL)
+		return FALSE;
+	if (!fu_udev_device_pread_full (aux_device, 0x50a, version_buf,
```

Comment:
is there a check you can do to make sure you're really talking to the LSPCON before you go and believe this offset means version?  Some sort of ID byte or anything?

Reply:
These bytes are defined to be version bytes by the DisplayPort specification, but I could also check the OUI at 0x500.

Reply:
OK.  I do think checking the OUI to know you're talking to the right lspcon is a good general safety thing then at least so when it comes time to rev the plugin for another parade part someone makes a conscious check on all the bits from this part to another.

Reply:
Added OUI check.